src/shims/time.rs

```diff
@@ -107,6 +112,87 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         Ok(0)
     }
 
+    // The localtime() function shall convert the time in seconds since the Epoch pointed to by
+    // timer into a broken-down time, expressed as a local time.
+    // https://linux.die.net/man/3/localtime_r
+    fn localtime_r(
+        &mut self,
+        timep: &OpTy<'tcx, Provenance>,
+        result_op: &OpTy<'tcx, Provenance>,
+    ) -> InterpResult<'tcx, Pointer<Option<Provenance>>> {
+        let this = self.eval_context_mut();
+
+        this.assert_target_os_is_unix("localtime_r");
+        this.check_no_isolation("`localtime_r`")?;
+
+        let timep = this.deref_pointer(timep)?;
+        let result = this.deref_pointer_as(result_op, this.libc_ty_layout("tm"))?;
+        let result_ptr = this.read_pointer(result_op)?;
+
+        let sec_since_epoch: i64 = this.read_scalar(&timep)?.to_i64()?;
```

Comment:
is it guaranteed to be pointer sized on all platforms? if so, you can use `read_target_isize` instead. Otherwise you can use `to_bits` instead of `to_i64`, but then you need to provide the size, which you can get from `timep.layout.size()`

Reply:
I am not sure how verify if it is pointer size in all platforms, but using ``read_target_isize`` did pass all the tests for tier 1 linux platform locally.

Reply:
For off_t we use something like
```rust
let offset = this.read_scalar(offset)?.to_int(this.libc_ty_layout("off_t").size)?;
```
Here it's `time_t`, I think; would make sense to use something similar.